specs/modal/readme.md

```diff
@@ -0,0 +1,164 @@
+| header-close-button | .header-close-button | the close button located in the header. Normally, at the top-right corner of the modal |
+| header-title | .header-title | the title of the modal found in the header |
+| body | .header-body | the body of the modal |
+| footer | .footer | the footer of the modal, usually containing at least two options: 1) to close or cancel the action, 2) to perform the primary action required by the modal (e.g., submit the form, login to the app) |
+| footer-close-button | .footer-close-button | the close or cancel button found in the footer |
+| footer-primary-button | .footer-primary-button | the primary button of the modal |
+
+Children with a role may also accept different props, such as a `title` for the child marked as header.
+
+### `onRequestClose` Sources
+
+| Source | Description |
+| --- | --- |
+| backdropClick | request to close the modal because the backdrop of the modal was clicked |
+| escKeyPress | request to close the modal because the `ESC` button was pressed |
+| headerCloseClick | request to close the modal came from a child in the role `header-close-button` |
+| footerCloseClick | request to close the modal came from a child in the role `footer-close-button` |
+| footerPrimaryClick | request to close the modal came from a child in the role `footer-primary-button`  |
+
+### Modal Header Props
+
+| Prop | Type | Default | Description |
+| --- | --- | --- | --- |
+| children | node | undefined | Alternative to using the text attribute, for when you need more control over the content|
+| onClose | func | undefined | What to do when the user clicks the close button|
+| title | string | '' | Creates a title for the modal |
+
+### Code Example
+
+**Example 1:**
+
+```jsx
+<Modal
+    isOpen={this.state.open}
+    onRequestClose={this.whenModalCloses}>
+    <div role={header}>
+        <span role={title}>Terms of service</span>
+        <button role={header-close-button} />
+    </div>
+    <div role={body}>
+        <span>content</span>
+    </div>
+    <div role={footer}>
+        <button role={footer-primary-button}>Yep</button>
+        <button role={footer-close-button}>Nope</button>
+    </div>
+</Modal>
+```
+
+## Style API
+
+Normally, the modal opens in relation to the screen and not its parent component. Because it breaks the flow of the entire page, it needs to be prominent on the entire view-port. 
+
+Its possible positions are top, **middle**, bottom and left, **center**, right (defaults bolded).
+
+It can have an offset from each of its non-default positions (top, bottom, right, left). The offset direction is opposite to its starting position (leftOffset pushes left to right, rightOffset pushes right to left).
+
+```jsx
+<Modal
+  position={left}
+  topOffset={10}
+  leftOffset={10}>
+</Modal>
+```
+
+#### Contained modals
+
+There are special cases where a modal is relevant only to a certain part of an application, and will only block its flow.
```

Comment:
This is very curious use-case. I don't recollect having seen this "in the wild". Also, might be hard to implement in React.